nilearn/_utils/testing.py

```diff
@@ -45,17 +46,56 @@ def assert_raises_regexp(expected_exception, expected_regexp,
             raise AssertionError("Should have raised %r" %
                                  expected_exception(expected_regexp))
 
+
+try:
+    from sklearn.utils.testing import clean_warning_registry
+except ImportError:
+    def clean_warning_registry():
+        """Safe way to reset warnings """
+        warnings.resetwarnings()
+        reg = "__warningregistry__"
+        for mod_name, mod in list(sys.modules.items()):
+            if 'six.moves' in mod_name:
+                continue
+            if hasattr(mod, reg):
+                getattr(mod, reg).clear()
+
+
 try:
     from sklearn.utils.testing import assert_warns
 except ImportError:
     # sklearn.utils.testing.assert_warns new in scikit-learn 0.14
     def assert_warns(warning_class, func, *args, **kw):
+        #clean_warning_registry()
         with warnings.catch_warnings(record=True):
-            warnings.simplefilter("ignore", warning_class)
+            warnings.simplefilter('always', warning_class)
```

Comment:
I believe this is a bug; I think it specifically suppresses `warning_class`, rather than forcing it to show (which is what I think we want.